plugins/plugin-docker/che-plugin-docker-machine/src/main/java/org/eclipse/che/plugin/docker/machine/DockerInstanceProvider.java

```diff
@@ -236,24 +298,18 @@ public Instance createInstance(Recipe recipe,
                               creationLogsOutput);
     }
 
-    @Override
-    public Instance createInstance(InstanceKey instanceKey,
-                                   Machine machine,
-                                   LineConsumer creationLogsOutput) throws NotFoundException, MachineException {
-        final DockerInstanceKey dockerInstanceKey = new DockerInstanceKey(instanceKey);
-        if (snapshotUseRegistry) {
-            pullImage(dockerInstanceKey, creationLogsOutput);
-        }
-        final String userName = EnvironmentContext.getCurrent().getSubject().getUserName();
-        final String machineContainerName = containerNameGenerator.generateContainerName(machine.getWorkspaceId(),
-                                                                                         machine.getId(),
-                                                                                         userName,
-                                                                                         machine.getConfig().getName());
+    protected Instance doCreateInstanceImage(final Machine machine, String machineContainerName,
```

Comment:
This method returns Instance which is representation of the container. Please rename this method.